marlowe-playground-client/src/MainFrame.purs

```diff
@@ -77,24 +80,29 @@ initialState =
     }
 
 ------------------------------------------------------------
-mainFrame ::
-  forall m.
+mkMainFrame ::
+  forall m n.
   MonadAff m =>
+  MonadEffect n =>
   MonadAsk (SPSettings_ SPParams_) m =>
-  Component HTML HQuery Unit WebsocketMessage m
-mainFrame =
-  H.mkComponent
-    { initialState: const initialState
-    , render
-    , eval:
-      H.mkEval
-        { handleQuery
-        , handleAction: handleActionWithAnalyticsTracking
-        , receive: const Nothing
-        , initialize: Just $ CheckAuthStatus
-        , finalize: Nothing
+  n (Component HTML HQuery Unit WebsocketMessage m)
```

Comment:
wouldn't it be better to do all this in `main` and then if we have other prefernces in the future they fit in there?

Reply:
Hmm...I'm not sure. I've tried to keep as much app-specific stuff as possible in `MainFrame`, and keep `Main` just about the mucky business of wiring the app into the browser and starting it up. 🤔 

By that logic though, perhaps we should be moving the `ajaxSettings` bit into `mkMainFrame` instead. What do you think?

Reply:
yes it should, they should both be together wherever that is

Reply:
Okay, done. :-)